### PR TITLE
Resolve various runtime errors.

### PR DESCRIPTION
**Pull request by timothyfroehlich**
# Based on PR #1947

## Chain of upstream PRs as of 2024-12-26

* PR #1886:
  `main` ← `feature/protoconv`

  * PR #1941:
    `feature/protoconv` ← `wb/froeht/protolize-dcf`

    * PR #1943:
      `wb/froeht/protolize-dcf` ← `wb/froeht/delete-proto-utils`

      * PR #1946:
        `wb/froeht/delete-proto-utils` ← `wb/froeht/temp-disable-variant-animation-test`

        * PR #1947:
          `wb/froeht/temp-disable-variant-animation-test` ← `wb/froeht/remove-serde-from-kotlin`

          * **PR #1948 (THIS ONE)**:
            `wb/froeht/remove-serde-from-kotlin` ← `wb/froeht/resolve-runtime-exceptions`

Protobuf maps are immutable and need to be modified by copying them.

### CODE REVIEW

File: designcompose/src/main/java/com/android/designcompose/VariableManager.kt [hunk 1]
```diff
@@ -170,20 +171,21 @@ internal object VariableManager {
...
-        varMap.variablesMap.putAll(map.variablesMap)
-        varMap.variableNameMapMap.putAll(map.variableNameMapMap)
-
+        varMap = varMap.copy {
+            // Remove old entries for docId
+            val oldVarMap = docVarMap[docId.id]
+            oldVarMap?.collectionsMap?.forEach { collections.remove(it.key) }
+            oldVarMap?.collectionNameMapMap?.forEach { collectionNameMap.remove(it.key) }
+            oldVarMap?.variablesMap?.forEach { variables.remove(it.key) }
+            oldVarMap?.variableNameMapMap?.forEach { variableNameMap.remove(it.key) }
```
Comment (rylin8) [inline designcompose/src/main/java/com/android/designcompose/VariableManager.kt:180]: This chunk doesn't need to be in the copy block does it?

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt [hunk 2]
```diff
@@ -566,24 +567,26 @@ private fun generateOverlayNode(
...
-            com.android.designcompose.definition.element.color {
-                r = (it.r * 255.0).toInt()
-                g = (it.g * 255.0).toInt()
-                b = (it.b * 255.0).toInt()
-                a = (it.a * 255.0).toInt()
-            }
-        nodeStyleBuilder.backgroundsList.add(background { solid = colorOrVar { color = bgColor } })
+    val newNodeStyle = node.style.nodeStyle.copy {
+        overflow = Overflow.OVERFLOW_VISIBLE
+        this.backgrounds.clear()
```
Comment (rylin8) [inline designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt:572]: I would keep the usage of `this` consistent. Either use it everywhere or not at all.

File: designcompose/src/main/java/com/android/designcompose/VariableManager.kt (hunk 1 above)
Comment (timothyfroehlich) [reply designcompose/src/main/java/com/android/designcompose/VariableManager.kt:180]: It does, it's modifying the varMap. I'll re-add the this's to make it more clear

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshLayout.kt [hunk 3]
```diff
@@ -296,8 +296,8 @@ internal fun layoutTree(
     val layoutParentChildren = arrayListOf<LayoutParentChildren>()
     updateLayoutTree(manager, root, layoutCache, layoutNodes, layoutParentChildren)
     val layoutNodeList = layoutNodeList {
-        layoutNodes
-        layoutParentChildren
+        this.layoutNodes.addAll(layoutNodes)
+        parentChildren.addAll(layoutParentChildren)
```
Comment (rylin8) [inline designcompose/src/main/java/com/android/designcompose/squoosh/SquooshLayout.kt:300]: can you add `this` here

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt [hunk 4]
```diff
@@ -566,24 +567,26 @@ private fun generateOverlayNode(
...
+                }
+            this.backgrounds.add(background { solid = colorOrVar { color = bgColor } })
+        }
     }
 
+
     val overlayStyle = viewStyle {
         layoutStyle = layoutStyleBuilder.build()
-        nodeStyle = nodeStyleBuilder.build()
+        nodeStyle = newNodeStyle
```
Comment (rylin8) [inline designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt:589]: can you add `this` to these

**Review by rylin8 — APPROVED** [review]

**Pull request merged**